### PR TITLE
County results displaying on hover

### PR DESCRIPTION
**Pull request by carlosm4247**
Description:
I took information from electionResults to build the county candidates and organized them into the table based on the vote count. The county results are simplified, only having candidate keys and votes. To get all the data I needed, I accessed the state vote candidates for their names and calculated the percentage. I used useState variables to keep track of the currently hovered county and the popup's position. The county popup updates with the current county's results using the ResultsBox component. The location of the popup also adjusts relative to the location of the county in the map. Finally, I created a border and some padding around the popup. 

Edit: Moved the border and padding styling from just the popup to all of the results boxes so they all have the same styling.

There are more styling things that could be done to it, but for now I'm focusing on the core features so I'm moving on to starting the feed in the next PR and will come back to do some more styling when I finish more of the core features.

Milestones:
- User can see election results for every state/district for past elections

Test Details:

https://github.com/carlosm4247/capstone-project/assets/126618515/d15d1ffb-4fa1-4c1c-bc22-da73549a9be4

### CODE REVIEW

**Comment by nick-r-117** [comment]
This looks great! Just quick feedback on UX: I think it would help to have a header/title to hover dialog to display which county the hover results is currently being rendered for. Also may help to increase the state size.

File: capstone-ui/src/Components/StateMap/StateMap.css [hunk 1]
```diff
@@ -2,4 +2,11 @@
     fill: #e3e3e3;
     stroke: #333;
     stroke-width: 0.5;
+}
+
```
Comment (nick-r-117) [inline capstone-ui/src/Components/StateMap/StateMap.css:6]: I think this commit could have been part of previous as last commit introduced the styling for `.popup-container`.

File: capstone-ui/src/Components/ResultsBox/ResultsBox.jsx [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import * as React from "react"
 import { useParams } from "react-router-dom"
 import electionResults from "../../Data/2020presidential.json"
+import "./ResultsBox.css"
```
Comment (nick-r-117) [inline capstone-ui/src/Components/ResultsBox/ResultsBox.jsx:4]: This commit could have also been apart of the previous.

File: capstone-ui/src/Components/ResultsBox/ResultsBox.jsx [hunk 3]
```diff
@@ -47,9 +47,20 @@ export default function ResultsBox( { locationLevel, countyFIPS } ) {
         }
         else if (locationLevel === 3) {
             data = electionResults.data.races.find( (race) => race.state_slug === stateName ).counties.find((county) => (county.fips%1000) === (countyFIPS%1000))
+
+            for (let candidate in data.results) {
```
Comment (nick-r-117) [inline capstone-ui/src/Components/ResultsBox/ResultsBox.jsx]: Do you have a limit to how much candidate results you would want to render in dialog? I expect not too many candidates per county, but to future proof a really long hover dialog with many candidates, you probably can add a max cap to how many candidates you find or use to build results to display.

**Review by nick-r-117 — APPROVED** [review]

**Comment by carlosm4247** [comment]
> This looks great! Just quick feedback on UX: I think it would help to have a header/title to hover dialog to display which county the hover results is currently being rendered for. Also may help to increase the state size.

Yes, I plan to work on making the counties/state names clear and adjusting the state bounds so the user can see it bigger and centered at some point after finishing technical challenges.

File: capstone-ui/src/Components/StateMap/StateMap.css (hunk 1 above)
Comment (carlosm4247) [reply capstone-ui/src/Components/StateMap/StateMap.css:6]: I'll keep that in mind

File: capstone-ui/src/Components/ResultsBox/ResultsBox.jsx (hunk 3 above)
Comment (carlosm4247) [reply capstone-ui/src/Components/ResultsBox/ResultsBox.jsx]: Not currently, but I do want to limit it and might take a look at it when I'm working on generating the candidate views.

**Pull request merged**